### ISSUE TITLE
Keyboard focus is not visible on 'tree view/filter/search' button in table.

### ISSUE THREAD

**Issue by sofisa16** [ag-grid/ag-grid#4948]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**
1. Go to https://www.ag-grid.com/react-data-grid/column-headers/#example-text-orientation
2. Click on the hamburger button on any header column:
![image](https://user-images.githubusercontent.com/14502698/156060794-97e9d51e-a23d-44dd-acbc-9511cd8b1627.png)
 3. This is what we see:
![image](https://user-images.githubusercontent.com/14502698/156061362-d3a12379-81fe-415c-981c-0787b1db4264.png)

**Expected behavior**
 4. However, this is what is expected to see:
![image](https://user-images.githubusercontent.com/14502698/156061289-78dd14be-5cc5-4da3-95d4-394cf8aad630.png)
 **6.  In other words, what is expected to see is a focus on the Search box.**

**Please tell us about your environment:**
I found the bug in https://www.ag-grid.com/react-data-grid/column-headers/#example-text-orientation

* **AG Grid version:** X.X.X
_According to https://www.ag-grid.com/react-data-grid/column-headers/#example-text-orientation , it is still an issue._

* **Browser:**
_'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36'_

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
_Thant info can be found https://www.ag-grid.com/react-data-grid/column-headers/#example-text-orientation_

**Comment by gportela85** [comment]
@sofisa16 This is a bit complicated to solve as the Text Field is not the first element in the panel (there is an expand / collapse button and a checkbox in front of it), so it works differently than the "Search" field in the Floating Filter.

If you do the same using the keyboard instead of a mouse
1 - Navigate to "Athlete" using arrows
2 - Press Ctrl + Enter to open the floating menu
3 - Press Shift + Tab to focus the tab header
4 - Press the right arrow and Tab

You will see that the focus is actually wrapped around the whole "Search" container like this: 

![Screen Shot 2022-03-01 at 09 49 29](https://user-images.githubusercontent.com/7908562/156201419-539a9aff-3801-4f4b-b67d-feab79a1e261.png)

if you are an Enterprise Customer, feel free to file an enhancement ticket for this issue, but at the moment is the best we can provide.
